### PR TITLE
Tweaking archer aiming and arrow bbox

### PR DESCRIPTION
**Pull request by Nananas**
## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Improves archer aiming and arrow bbox. Will make it easier to shoot through gaps, and will be less confusing for (new) players.

- moved archer bow 2px up to align with arrow spawn. This was originally chosen to make it easy to shoot over level blocks, now the bow should be on the same level.
- align *arrow spawn* better when aiming up/down. The bow animation is out-of-center horizontally. This is most notabily when aiming up and down
- arrow bbox now rectangle to fit better in between blocks

## Steps to Test or Reproduce

Mostly try shooting in between small gaps. The arrow on the bow animation should be a better indicator if you can shoot through. Water arrows should still stun behind 2 layer walls.

### CODE REVIEW

File: Entities/Characters/Archer/ArcherLogic.as [hunk 1]
```diff
@@ -797,7 +797,16 @@ void ClientFire(CBlob@ this, const s8 charge_time, const bool hasarrow, const u8
...
+		Vec2f offset;
+		if (this.isFacingLeft()) 
+		{
+			offset.x = 2;
+		} 
+		else
+		{
+			offset.x = -2;
+		}
+		ShootArrow(this, this.getPosition() + Vec2f(0.0f, -2.0f) + offset, this.getAimPos(), arrowspeed, arrow_type, legolas);
```
Comment (eps0003) [inline Entities/Characters/Archer/ArcherLogic.as:809]:
```suggestion
		Vec2f offset(this.isFacingLeft() ? 2 : -2, -2);
		ShootArrow(this, this.getPosition() + offset, this.getAimPos(), arrowspeed, arrow_type, legolas);
```

File: Entities/Characters/Archer/ArcherAnim.as [hunk 2]
```diff
@@ -235,8 +235,13 @@ void onTick(CSprite@ this)
 	bool crouch = false;
 
 	bool knocked = isKnocked(blob);
-	Vec2f pos = blob.getPosition();
+	Vec2f pos = blob.getPosition() + Vec2f(0, -2);
 	Vec2f aimpos = blob.getAimPos();
+	if (this.isFacingLeft())
+		pos.x += 2;
+	else
+		pos.x -= 2;
```
Comment (eps0003) [inline Entities/Characters/Archer/ArcherAnim.as:243]:
```suggestion
	pos.x += this.isFacingLeft() ? 2 : -2;
```

**Review by eps0003 — COMMENTED** [review]
Some minor suggestions

**Comment by Nananas** [comment]
Looks much cleaner now!

**Review by eps0003 — APPROVED** [review]
Acts as expected when I did a quick test 👍

**Review by AsuMagic — APPROVED** [review]

**Pull request merged**